Clear up and fix tutorials

benfitzpatrick opened the pull request:
This clears up some bits of the 3 tutorials (hello world, cycling, and suite writing) based on going through it on the VM with a confused-user hat on.

There are a couple of fixes for the suite writing tutorial too.

@oliver-sanders, please review.

benfitzpatrick committed:
Alter tutorials based on going through them

benfitzpatrick committed:
Add .gitignore

benfitzpatrick committed:
Add hello world and rocket suites into the distribution

benfitzpatrick commented:
Suite files added here

hjoliver commented on suite-writing.tex:
```diff
@@ -20,7 +20,7 @@ \subsection{Starting Out}
 Our Fortran program reads in some position data and then pretends to calculate a new one based on a compass direction and a 5 knot speed for a given period of time.
 
-If you are reading this tutorial via the 'metomi VM' then the Fortran code is located at {\em \$HOME/tutorial/suites/navigation/src/dead\_reckoning.f90}. Otherwise, the code is given in Appendix~\ref{Appendix Suite Writing Tutorial}.
+If you are reading this tutorial via the 'metomi VM' then the Fortran code is located at {\em \$HOME/tutorial/suites/navigation/src/dead\_reckoning.f90}. Otherwise, the code is given in Appendix~\ref{Appendix In Depth Suite Writing Tutorial}.
```
Calling it the "cylc VM" above.

hjoliver commented on suites/navigation/suite.rc:
```diff
@@ -0,0 +1,16 @@
+            graph = compile_navigate
+[runtime]
+    [[root]]
+        pre-script = sleep 5 # Slow down tasks for visualization.
```
Is 5 seconds enough of a slowdown here?

hjoliver commented:
Should we provide completed versions of the incrementally-developed suites too, for reference (e.g. in case the user screws up and can't figure out what's wrong)?

hjoliver commented:
In the cycling example suite, could you write the graph like this (or similar):

```
graph = """point_upwards => load_astronauts & fill_fuel_tank & set_coordinates
                       fill_fuel_tank => light_fuse
           set_coordinates & load_astronauts & light_fuse  => count_down => blast_off
                       blast_off[-P1D] => point_upwards"""
```

To my mind this is easier to interpret and more intuitive than a long list of individual dependencies for which the reader has to work out what runs concurrently (whereas the `&`s here make this obvious).

hjoliver commented:
I think "In Depth..." should be hyphenated when followed by a noun phrase like "suite writing tutorial".  (But not when followed by a verb, so _I am studying this topic in depth, in order to write and in-depth review_.)

hjoliver commented:
The yellow of the ship flag is not quite the same shade as the  yellow in the cylc logo. :boom:

hjoliver commented:
Section 4:
- what's an "existing standard template suite"  - might be better to say "start from an existing suite"?
- need to explain what `$CYLC_SUITE_SHARE_DIR` is?  Also, inconsistent use of _DIR and _PATH.
- nice work on the Google Maps integration :grin: (I have never gone through the rose tutorial suites in detail before, although I've skimmed through quickly and admired them from a distance)
- one use of old terminology "cycle time" detected (sec 4.8)
- sec 4.11 margin over-run in replacement scripting for write_start_position
- (I didn't know about `rose metadata-gen` -- nice!)
- in 4.11 when you move stuff out of suite.rc into a rose app config, maybe explain why that is a good thing for the benefit of those whose initial reaction will be "how does that help - you've just hidden some important configuration that was nice and clear in the suite.rc before") - i.e. the suite definition describes the _workflow_ and any shared variables needed to make all the tasks work together, anything task-specific does not need to be there; plus benefits of rose app metadata...
- maybe add a nice screenshot of `rose edit` (could be in an appendix) to entice those who otherwise might not be bothered going through the optional Rose section?
- highlight "This is only a limited subset of Rose functionality..." in a `\note{...}` paragraph

benfitzpatrick committed:
Fix colour of ship flag!

benfitzpatrick commented:
> I think "In Depth..." should be hyphenated when followed by a noun phrase like "suite writing tutorial". 

Fair enough, agreed - I do wonder if that whole phrase is a good name for the tutorial in this context.

oliver-sanders commented:
@hjoliver 

> In the cycling example suite, could you write the graph like this (or similar):

The reason that I didn't do it like this was because we haven't introduced the `&` / `|` notation yet (see advanced tutorials). I appreciate that the syntax is pretty self explanatory, my preference is to keep the tutorial to the point then introduce extras afterwards.

benfitzpatrick committed:
Address @hjoliver feedback

benfitzpatrick commented:
Ok, I think I've addressed all your comments. A screenshot of `rose edit` and corresponding `rose-app.conf` snippet is on its way pending UM code owner permission (I thought a real screenshot might be best).

hjoliver commented:
@oliver-sanders -

> we haven't introduced the & / | notation yet ...

Not true - unless you've reversed my earlier changes under _Cylc Introduction_ -> _Dependency Graph_.    I don't agree that '&' should be viewed as an "extra" advanced feature (presumably for making graphs conveniently but unnecessarily more concise or something?) because its meaning is pretty obvious _and_ not using it actually makes a graph harder to understand in the sense that concurrency and structure (c.f. the picture generated by "cylc graph") has to be reconstructed in the user's head (as opposed to staring her in the face!).

benfitzpatrick commented:
Now mergable. Cycling example suite TBD in a different PR?

hjoliver commented:
Changes look good, although you didn't address my comment on not writing graphs as a list of individual dependencies, given that we have in fact introduced AND and OR operators earlier (just asking... maybe you both prefer the list layout?).

benfitzpatrick committed:
Condense cycling example graphing

benfitzpatrick commented:
Did you not just mean Oliver's cycling example? I had thought we could do that in another PR, but I've put a suggestion up in be4e1b7.

hjoliver commented:
> Did you not just mean Oliver's cycling example?

I assume you mean  "list of dependencies" vs use of chaining and AND for a more "graphy" graph string.  I would prefer the latter in all cases, but it's a pretty minor issue, and subjective ... I've made my point and will leave it up to you guys now.

benfitzpatrick committed:
Add rose edit screenshot

benfitzpatrick committed:
Remove need for continuation

benfitzpatrick commented:
OK, added rose edit screenshot and cleared up continuation line. @oliver-sanders, merge please?
